amd_openvx_extensions/amd_rpp/source/Downmix.cpp

```diff
@@ -0,0 +1,271 @@
+{
+    DownmixLocalData *data = new DownmixLocalData;
+    memset(data, 0, sizeof(*data));
+    STATUS_ERROR_CHECK(vxCopyScalar((vx_scalar)parameters[5], &data->device_type, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    STATUS_ERROR_CHECK(vxReadScalarValue((vx_scalar)parameters[4], &data->nbatchSize));
+
+    // Querying for input tensor
+    data->src_desc_ptr = &data->srcDesc;
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_NUMBER_OF_DIMS, &data->src_desc_ptr->numDims, sizeof(data->src_desc_ptr->numDims)));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DIMS, &data->in_tensor_dims, sizeof(vx_size) * data->src_desc_ptr->numDims));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DATA_TYPE, &data->in_tensor_type, sizeof(data->in_tensor_type)));
+    if (data->in_tensor_type == vx_type_e::VX_TYPE_FLOAT32)
+        data->src_desc_ptr->dataType = RpptDataType::F32;
+    data->src_desc_ptr->offsetInBytes = 0;
+
+    // Querying for output tensor
+    data->dst_desc_ptr = &data->dstDesc;
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_NUMBER_OF_DIMS, &data->dst_desc_ptr->numDims, sizeof(data->dst_desc_ptr->numDims)));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_DIMS, &data->out_tensor_dims, sizeof(vx_size) * data->dst_desc_ptr->numDims));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1],VX_TENSOR_DATA_TYPE, &data->out_tensor_type, sizeof(data->out_tensor_type)));
+    if (data->out_tensor_type == vx_type_e::VX_TYPE_FLOAT32)
+        data->dst_desc_ptr->dataType = RpptDataType::F32;
+    data->dst_desc_ptr->offsetInBytes = 0;
+
+    // source_description_ptr
+    data->src_desc_ptr->n = data->in_tensor_dims[0];
+    data->src_desc_ptr->h = data->in_tensor_dims[2];
+    data->src_desc_ptr->w = data->in_tensor_dims[1];
+    data->src_desc_ptr->c = 1;
+    data->src_desc_ptr->strides.nStride = data->src_desc_ptr->c * data->src_desc_ptr->w * data->src_desc_ptr->h;
+    data->src_desc_ptr->strides.hStride = data->src_desc_ptr->c * data->src_desc_ptr->w;
+    data->src_desc_ptr->strides.wStride = data->src_desc_ptr->c;
+    data->src_desc_ptr->strides.cStride = 1;
+    data->src_desc_ptr->numDims = 4;
+
+    // source_description_ptr
+    data->dst_desc_ptr->n = data->in_tensor_dims[0];
+    data->dst_desc_ptr->w = data->in_tensor_dims[1];
+    data->dst_desc_ptr->h = 1;
+    data->dst_desc_ptr->c = 1;
+    data->dst_desc_ptr->strides.nStride = data->dst_desc_ptr->c * data->dst_desc_ptr->w * data->dst_desc_ptr->h;
+    data->dst_desc_ptr->strides.hStride = data->dst_desc_ptr->c * data->dst_desc_ptr->w;
+    data->dst_desc_ptr->strides.wStride = data->dst_desc_ptr->c;
+    data->dst_desc_ptr->strides.cStride = 1;
+    data->dst_desc_ptr->numDims = 4;
+    data->samples = (vx_int32 *)malloc(sizeof(vx_int32) * data->src_desc_ptr->n);
```

Comment:
Is it possible to use std::vector here instead of mallocs in the current file?
I know this is is similar to existing openvx files, but check if there is possibility of using vector instead of malloc